ddp-workspace/projects/ddp-sdk/src/lib/components/user/activities/userActivities.component.ts

```diff
@@ -201,10 +202,13 @@ import { take } from 'rxjs/operators';
 })
 export class UserActivitiesComponent implements OnInit, OnDestroy, OnChanges, AfterContentInit {
     @Input() studyGuid: string;
+    // in case we use multiple ddp-user-activities component instances on the page we can't use session participantGuid
+    @Input() participantGuid: string;
+    @Input() activities: Array<ActivityInstance>;
     @Input() displayedColumns: Array<DashboardColumns> = ['name', 'summary', 'date', 'status', 'actions'];
     @Output() open: EventEmitter<string> = new EventEmitter();
     @Output() loadedEvent: EventEmitter<boolean> = new EventEmitter<boolean>();
-    public dataSource: UserActivitiesDataSource;
+    public dataSource: UserActivitiesDataSource | Array<ActivityInstance>;
```

Comment:
Does it make sense in some cases getting the activities from and input parameter vs in other cases getting them from a datasource? Would it be clearer if activities to be displayed came from one clear path?
(I am surprised the table will take an array or a datasource! Had to look it up in APIs)

Reply:
Actually it's a very good suggestion, I liked it and thought about that, but my impression was we used it in much places and it would be hard to refactor, but in fact we use it only in toolkit dashboard components. So done!